event-sourcing/event-store-api/src/main/scala/org/apache/james/eventsourcing/eventstore/EventStore.scala

```diff
@@ -7,37 +7,35 @@
  * "License"); you may not use this file except in compliance   *
  * with the License.  You may obtain a copy of the License at   *
  *                                                              *
- *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
  *                                                              *
  * Unless required by applicable law or agreed to in writing,   *
  * software distributed under the License is distributed on an  *
  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
  * KIND, either express or implied.  See the License for the    *
  * specific language governing permissions and limitations      *
  * under the License.                                           *
- ****************************************************************/
+ * ***************************************************************/
+package org.apache.james.eventsourcing.eventstore
 
-package org.apache.james.eventsourcing;
+import org.apache.james.eventsourcing.{AggregateId, Event}
 
-import java.util.List;
+import scala.annotation.varargs
+import scala.jdk.CollectionConverters._
 
-public interface Event extends Comparable<Event> {
+trait EventStore {
```

Comment:
why did you converted this interface? Isn't it easier to have interfaces for java implementations written in java?

Reply:
All the implementation of this Trait are now scala classes which is one of the goal of this PR.